Release/include/pplx/pplxtasks.h

```diff
@@ -4391,11 +4408,11 @@ class task<void>
     ///     Runtime)"/>.</para>
     /// </remarks>
     /**/
-    template<typename _Ty>
+    template<typename _Ty, typename = typename details::_EnableIfNotTaskRet<typename std::decay<_Ty>::type, void>>

```

Comment:
This is exactly what `_EnableIfNotTaskRet` doing: checks that `_Ty` is not derived from `task<_ReturnType>` (`is_base_of` check includes `is_same` logic in this case). So we can catch that tricky case when `_Ty` is not a `task` class itself, but its child class, when compiler may decide to use this forwarding-reference overload instead of copy- or move-constructor.